src/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -26,64 +25,66 @@ public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSett
         {
             try
             {
-                var model = JObject.Parse(launchSettingsJsonContents);
-                var profilesObject = model[ProfilesKey] as JObject;
+                var model = JsonDocument.Parse(launchSettingsJsonContents).RootElement;
 
-                if (profilesObject == null)
+                if (model.Type != JsonValueType.Object || !model.TryGetProperty(ProfilesKey, out var profilesObject) || profilesObject.Type != JsonValueType.Object)
                 {
                     return new LaunchSettingsApplyResult(false, LocalizableStrings.LaunchProfilesCollectionIsNotAJsonObject);
                 }
 
-                JObject profileObject;
+                JsonElement profileObject;
                 if (profileName == null)
                 {
                     profileObject = profilesObject
-                        .Properties()
-                        .FirstOrDefault(IsDefaultProfileType)?.Value as JObject;
+                        .EnumerateObject()
+                        .FirstOrDefault(IsDefaultProfileType).Value;
                 }
                 else
                 {
-                    profileObject = profilesObject[profileName] as JObject;
-
-                    if (profileObject == null)
+                    if (!profilesObject.TryGetProperty(profileName, out profileObject) || profileObject.Type != JsonValueType.Object)
                     {
                         return new LaunchSettingsApplyResult(false, LocalizableStrings.LaunchProfileIsNotAJsonObject);
                     }
                 }
 
-                if (profileObject == null)
+                if (profileObject.Type == default)
```

Comment:
Can this if evaluate to true?  If no property matched `IsDefaultProfileType` then the previous `if` would have thrown when accessing the Value property of the nullable.

If a profile was named but the name was not found or the name didn't point to an object then it already returned.

So in either case it seems like profileObject should be assigned, and not of type Undefined.

Reply:
If I'm following correctly, the `.Value` access is on the FirstOrDefault return of `JsonProperty` (non-nullable). In the case that FirstOrDefault didn't find a match, this should be `default(JsonProperty)` rather than a null `Nullable<JsonProperty>`, so the access shouldn't throw - it should produce a `default(JsonElement)` with a `Type` property also set to the default value (in this case `JsonValueType.Undefined`).